python/spacewalk/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -681,7 +683,8 @@ def _prep_zypp_repo_url(self, url, uln_repo):
         if self.sslclientkey:
             query_params['ssl_clientkey'] = self.sslclientkey
         new_query = unquote(urlencode(query_params, doseq=True))
-        if self.authtoken or uln_repo:
+        query_check = urlparse(url)
+        if (query_check.query != "") or uln_repo:
```

Comment:
I'm not really sure about this part. Why not just let `urlunparse` do it's job to compose the URL back?

Reply:
You're right, rewriting this part to use unparse might be the better way to go than keeping these checks with weird logic.  I'll change it